frontend/lib/identification/card_detail_view/card_detail_view.dart

```diff
@@ -107,6 +108,7 @@ class RichQrCode extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final bool isExpired = isCardExpired(userCode.info);
```

Comment:
Minor: I think `isExpired` should be a "prop" for `RichQrCode`. Then we only evaluate `isCardExpired` once in `CardDetailView`.